It looks like we still have a chicken and egg problem.
```
[ERROR] Failed to execute goal org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:5.2.10.Final:build-feature-pack (feature-pack-build) on project wildfly-ee-galleon-pack: Package org.jboss.ws.cxf.jbossws-cxf-server has unsatisifed external dependency on package org.picketbox -> [Help 1]
[ERROR] Failed to execute goal org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin:5.2.10.Final:build-feature-pack (feature-pack-build) on project wildfly-preview-feature-pack: Package org.jboss.ws.cxf.jbossws-cxf-server has unsatisifed external dependency on package org.picketbox -> [Help 1]
```

--- a/ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/as/webservices/server/integration/main/module.xml
+++ b/ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/as/webservices/server/integration/main/module.xml
@@ -81,6 +81,7 @@
             <include path="META-INF"/>
           </exports>
         </module>
+        <module name="org.apache.commons.codec"/>
         <module name="org.apache.xalan" services="export" export="true"/>
         <module name="org.apache.xerces" services="export" export="true"/>
         <module name="org.jboss.as.webservices" services="export" export="true"/>
@@ -95,5 +96,6 @@
             <include path="org.jboss.jbossts.txbridge.outbound"/>
           </imports>
         </module>
+        <module name="org.wildfly.security.elytron"/>
     </dependencies>
 </module>

--- a/ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-server/main/module.xml
+++ b/ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-server/main/module.xml
@@ -43,6 +43,7 @@
         <module name="javax.xml.bind.api" />
         <module name="javax.xml.stream.api" />
         <module name="javax.xml.ws.api" />
+        <module name="org.apache.commons.codec" />
         <module name="org.jboss.ws.api" />
         <module name="org.jboss.ws.spi" />
         <module name="org.jboss.ws.common" />
@@ -69,6 +70,7 @@
         </module>
         <module name="org.jboss.logging" />
         <module name="org.apache.ws.security" />
+        <module name="org.wildfly.security.elytron" />
         <module name="java.xml"/>
     </dependencies>
 </module>

--- a/pom.xml
+++ b/pom.xml
@@ -474,7 +474,7 @@
         <version.org.jboss.ws.common.tools>1.3.2.Final</version.org.jboss.ws.common.tools>
         <version.org.jboss.ws.cxf>5.4.4.Final</version.org.jboss.ws.cxf>
         <version.org.jboss.ws.jaxws-undertow-httpspi>1.0.1.Final</version.org.jboss.ws.jaxws-undertow-httpspi>
-        <version.org.jboss.ws.spi>3.3.1.Final</version.org.jboss.ws.spi>
+        <version.org.jboss.ws.spi>3.4.0.Final</version.org.jboss.ws.spi>
         <version.org.jboss.xnio.netty.netty-xnio-transport>0.1.9.Final</version.org.jboss.xnio.netty.netty-xnio-transport>
         <version.org.jctools.jctools-core>2.1.2</version.org.jctools.jctools-core>
         <version.org.jgroups>4.2.18.Final</version.org.jgroups>

--- a/webservices/server-integration/src/main/java/org/jboss/as/webservices/security/ElytronSecurityDomainContextImpl.java
+++ b/webservices/server-integration/src/main/java/org/jboss/as/webservices/security/ElytronSecurityDomainContextImpl.java
@@ -57,6 +57,12 @@ public class ElytronSecurityDomainContextImpl implements SecurityDomainContext {
     public String getSecurityDomain() {
         return this.securityDomain.toString();
     }
+
+    @Override
+    public SecurityDomain getElytronSecurityDomain() {
+        return this.securityDomain;
+    }
+
     //TODO:refactor/deprecate this after elytron?
     @Override
     public Set<Principal> getUserRoles(Principal principal) {

